### PR TITLE
test: added test for correct instantiation of nested function properties

### PR DESCRIPTION
**Pull request by timoklingenhoefer**
Refers to class-transformer bug:
https://github.com/typestack/class-transformer/issues/972
https://github.com/typestack/class-transformer/issues/876

### CODE REVIEW

**Comment by 3fd** [comment]
can you add a test for the bug and link to the issue that describes the bug?

File: packages/sdk/test/flow.spec.ts [hunk 1]
```diff
@@ -260,6 +261,53 @@ describe('Flow Application', () => {
 
     expect(loggerMock.log).toHaveBeenCalledWith('Flow Deployment is running', expect.objectContaining(flow.context));
   }, 60000);
+
+  it('FLOW.FA.9 test complex properties', (done) => {
```
Comment (3fd) [inline packages/sdk/test/flow.spec.ts:265]: the test succeeds wether enableImplicitConversion is set to true or false

Comment (timoklingenhoefer) [reply packages/sdk/test/flow.spec.ts:265]: thats correct for some reason it always works in this repo, but in the hpc it does not work without the parameter

Comment (3fd) [reply packages/sdk/test/flow.spec.ts:265]: The reason was that the wrong class-transformer version was used in the flow modules repo. Installing with the "legacy-peer-deps" flag resolved it.
You can remove the workaround if you want to merge the test anyways

Comment (3fd) [reply packages/sdk/test/flow.spec.ts:265]: Can you please also remove the changeset and update the PR title?

File: .changeset/selfish-emus-melt.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@hahnpro/flow-sdk': patch
+---
+
+added test for correct instantiation for nested function properties
```
Comment (3fd) [inline .changeset/selfish-emus-melt.md:5]: Any particular reason why you want to release this as a new version? No features or fixes are added

**Pull request merged**